sphinx_gallery/tests/test_full.py

```diff
@@ -522,32 +527,67 @@ def test_rebuild(tmpdir_factory, sphinx_app):
+        # modify must be last as this rerun setting tries to run the
+        # broken example (subsequent tests depend on it)
+        _rerun(how, src_dir, conf_dir, out_dir, toctrees_dir,
+               generated_modules_0, generated_backrefs_0, generated_rst_0,
+               generated_pickle_0, copied_py_0, copied_ipy_0)
+
+
+def _rerun(how, src_dir, conf_dir, out_dir, toctrees_dir,
+           generated_modules_0, generated_backrefs_0, generated_rst_0,
+           generated_pickle_0, copied_py_0, copied_ipy_0):
+    """Rerun the sphinx build and check that the right files were changed."""
     time.sleep(0.1)
-    fname = op.join(src_dir, 'examples', 'plot_numpy_matplotlib.py')
-    with codecs.open(fname, 'r', 'utf-8') as fid:
-        lines = fid.readlines()
-    with codecs.open(fname, 'w', 'utf-8') as fid:
-        for line in lines:
-            if line.startswith('FYI this'):
-                line = 'A ' + line
-            fid.write(line)
+    confoverrides = dict()
+    if how == 'modify':
+        fname = op.join(src_dir, 'examples', 'plot_numpy_matplotlib.py')
+        with codecs.open(fname, 'r', 'utf-8') as fid:
+            lines = fid.readlines()
+        with codecs.open(fname, 'w', 'utf-8') as fid:
+            for line in lines:
+                if 'FYI this' in line:
+                    line = 'A ' + line
+                fid.write(line)
+        out_of, excluding = N_FAILING + 1, N_GOOD - 1
+        n_stale = N_GOOD - 1
+    else:
+        assert how == 'run_stale'
+        confoverrides['sphinx_gallery_conf.run_stale_examples'] = 'True'
+        confoverrides['sphinx_gallery_conf.filename_pattern'] = 'plot_numpy_ma'
+        out_of, excluding = 1, 0
+        n_stale = 0
     with docutils_namespace():
         new_app = Sphinx(src_dir, conf_dir, out_dir, toctrees_dir,
-                         buildername='html', status=StringIO())
+                         buildername='html', status=StringIO(),
+                         confoverrides=confoverrides)
         new_app.build(False, [])
     status = new_app._status.getvalue()
-    n = '[' + '|'.join(str(x + N_FAILING) for x in range(4)) + ']'
     lines = [line for line in status.split('\n') if 'source files tha' in line]
-    want = '.*targets for %s source files that are out of date$.*' % n
-    assert re.match(want, status, re.MULTILINE | re.DOTALL) is not None, lines
+    lines = '\n'.join([how] + lines)
+    flags = re.MULTILINE | re.DOTALL
+    # for some reason, setting "confoverrides" above causes Sphinx to show
+    # all targets out of date, even though they haven't been modified...
+    if how == 'run_stale':
+        want = '.*targets for %s source files that are out of date$.*' % N_RST
+        assert re.match(want, status, flags) is not None, lines
+    # ... but then later detects that only some have actually changed
+    # (sometimes this is 8, other times 9, depending on Windows/Linux for some
+    # unknown reason)
```

Comment:
For me 'run_stale' gives '8 changed' and 'modify' gives '9 changed' - which seems to make sense as we modify a file in 'modify'...?

Reply:
The reason I had to put `[8|9]` was from 5164de3 -- Travis (Linux) passed with 9 but AppVeyor (Windows) wanted 8.